service/src/main.rs

```diff
@@ -592,68 +597,94 @@ fn start_worker<E, T, D, InitializationHandler, WorkerModeProvider>(
 	}
 
 	if let Some(url) = config.target_a_parentchain_rpc_endpoint() {
-		init_target_a_parentchain(&enclave, &tee_accountid, url, is_development_mode)
+		init_target_parentchain(
+			&enclave,
+			&tee_accountid,
+			url,
+			ParentchainId::TargetA,
+			is_development_mode,
+		)
+	}
+
+	if let Some(url) = config.target_b_parentchain_rpc_endpoint() {
+		init_target_parentchain(
+			&enclave,
+			&tee_accountid,
+			url,
+			ParentchainId::TargetB,
+			is_development_mode,
+		)
 	}
 
 	// ------------------------------------------------------------------------
 	// Subscribe to events and print them.
-	println!("*** Subscribing to events");
+	println!("*** [{:?}] Subscribing to events", ParentchainId::Integritee);
 	let mut subscription = integritee_rpc_api.subscribe_events().unwrap();
-	println!("[+] Subscribed to events. waiting...");
+	println!("[+] [{:?}] Subscribed to events. waiting...", ParentchainId::Integritee);
 	loop {
 		if let Some(Ok(events)) = subscription.next_event::<RuntimeEvent, Hash>() {
 			print_events(events)
 		}
 	}
 }
 
-fn init_target_a_parentchain<E>(
+fn init_target_parentchain<E>(
 	enclave: &Arc<E>,
 	tee_account_id: &AccountId32,
 	url: String,
+	parentchain_id: ParentchainId,
 	is_development_mode: bool,
 ) where
 	E: EnclaveBase + Sidechain,
 {
+	println!("Initializing parentchain {:?} with url: {}", parentchain_id, url);
 	let node_api = NodeApiFactory::new(url, AccountKeyring::Alice.pair())
 		.create_api()
-		.expect("Failed to create Target A parentchain node API");
+		.unwrap_or_else(|_| panic!("[{:?}] Failed to create parentchain node API", parentchain_id));
```

Comment:
Using a `format!` in an expect raises a clippy warning as non-default, non-static code paths should better be invoked in a closure.